src/main/java/org/candlepin/controller/EntitlementCertificateGenerator.java

```diff
@@ -328,6 +333,122 @@ public void regenerateCertificatesOf(Consumer consumer, boolean lazy) {
         this.regenerateCertificatesOf(consumer.getEntitlements(), lazy);
     }
 
+    @Transactional
+    public void regenerateCertificatesOf(Consumer consumer, List<String> updatedEnvironments,
+        List<String> environmentChangeList, boolean lazy) {
+        log.info(
+            "Regenerating #{}, entitlement certificates for consumer: {}",
+            consumer.getEntitlements().size(), consumer
+        );
+
+        // we need to clear the content access cert on regenerate
+        Owner owner = ownerCurator.findOwnerById(consumer.getOwnerId());
+        if (owner.isUsingSimpleContentAccess()) {
+            this.contentAccessManager.removeContentAccessCert(consumer);
+        }
+
+        List<Entitlement> entitlements = filterEntitlements(consumer,
+            updatedEnvironments, environmentChangeList);
+
+        // TODO - Assumes only 1 entitlement certificate exists per entitlement
+        this.regenerateCertificatesOf(entitlements, lazy);
+    }
+
+    /**
+     * Select consumers entitlement whose entitlement certs need to be
+     * regenerated.
+     *
+     * @param consumer
+     *  The consumer for which to regenerate entitlement certificates
+     *
+     * @param updatedEnvironments
+     *  List of updated environment
+     *
+     * @param environmentChangeList
+     *  List of environment that gets changed
```

Comment:
Can we explain a bit more what this field is supposed to be? The explanation doesn't look much different from the one above. If it's about environments that got "added, removed, or reprioritized", then we need to explain it here. If we could find a better name for the variable it would be great, but I can't think of something better, so this is fine (as long as the javadoc explains in detail)